src/FormidableValidations.res

```diff
@@ -27,34 +27,36 @@ module Validator = {
 }
 
 module Description = {
-  type t<'values, 'value, 'error, 'label> = {
-    names: list<'label>,
-    validator: Validator.t<'values, 'value, 'error>,
-  }
+  type kind<'label> = [#name('label) | #names(array<'label>)]
+
+  let resolveKind = (kind: kind<'label>) =>
+    switch kind {
+    | #name(label) => [label]
+    | #names(labels) => labels
+    }
+
+  type t<'values, 'value, 'error, 'label> = (kind<'label>, Validator.t<'values, 'value, 'error>)
 }
 
 type t<'values, 'value, 'error, 'label> = (
   Strategy.t,
   Description.t<'values, 'value, 'error, 'label>,
 )
 
-let compose = (
-  {Description.names: names, validator},
-  {Description.names: names', validator: validator'},
-) => {
-  Description.names: names->List.concat(names'),
-  validator: field =>
+let compose = ((names, validator), (names', validator')) => (
+  names->Description.resolveKind->Js.Array2.concat(names'->Description.resolveKind)->#names,
+  field =>
     switch validator(field) {
     | #ok(_) => validator'(field)
     | #error(_) as error => error
     },
-}
+)
 
 let getStrategy = ((strategy, _)) => strategy
 
-let getNames = ((_, {Description.names: names})) => names
+let getNames = ((_, (names, _))) => names->Description.resolveKind
 
-let getValidator = ((_, {Description.validator: validator})) => validator
+let getValidator = ((_, (_, validator))) => validator
```

Comment:
so we will use tuple in the end and not convert to object @gaku-sei -san?

Reply:
ah I see just read the description ! sorry :P

Reply:
Only for this reason:

>No more a record, since we have #name it's probably explicit enough, and it prevents this: names: #name(#required) which is a bit weird

It's really weird to define a validation with `names: name` 😕 

But I don't mind to bring them back ^^

Reply:
since you are using kind like above, I think it is clear enough :D. And this one looks simpler than 22 so probably good with it